cpp/src/remote_handle.cpp

```diff
@@ -17,11 +17,6 @@
 #include <cassert>
 #include <cstddef>
 #include <cstring>
-#include <iostream>
-#include <memory>
```

Comment:
`<memory>` is needed for `std::unique_ptr`.

Reply:
The header file `remote_handle.hpp` has already included `<memory>` for the unique pointers used in the header.

Reply:
IIUC, we should include all used headers, regardless of redundancy with what the included headers include themselves.

Reply:
The headers I removed were reported by the `clangd` plug-in that comes with the dev container. The plug-in uses the "[include cleaner](https://clang.llvm.org/extra/clang-tidy/checks/misc/include-cleaner.html)" to identify the unused include. Given that transitive include is discouraged, that we do not have `misc-include-cleaner` in the `.clang-tidy` file, and that, more importantly, the reported warnings are mysteriously gone now (vscode bug?!), I'll revert the changes.

Reply:
Yes, we aim to avoid relying transitive header inclusions. If an identifier is used in a file, the corresponding header should be included.

Reply:
That said, I am happy to accept partial fixes that improve the state of things. For cleanups like this, it is easy to let work drag on. I prefer to accept improvements as-is even if they don’t fix every instance of a problem. Then future diffs can be smaller and we avoid generating conflicts along the way.